### PR TITLE
[Kotlin Migration] FinderTest

### PR DESCRIPTION
**Pull request by n1snt**
## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented on your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt [hunk 1]
```diff
@@ -13,223 +13,257 @@
...
-        assertEquals(1, col.findCards("tag:monkey").size());
-        assertEquals(1, col.findCards("tag:sheep -tag:monkey").size());
-        assertEquals(4, col.findCards("-tag:sheep").size());
-        col.getTags().bulkAdd(col.getDb().queryLongList("select id from notes"), "foo bar");
-        assertEquals(5, col.findCards("tag:foo").size());
-        assertEquals(5, col.findCards("tag:bar").size());
-        col.getTags().bulkRem(col.getDb().queryLongList("select id from notes"), "foo");
-        assertEquals(0, col.findCards("tag:foo").size());
-        assertEquals(5, col.findCards("tag:bar").size());
+        Assert.assertEquals(5, col.findCards("tag:*").size.toLong())
```
[the remark this answers is not in the document]
Comment (Arthur-Milchior) [reply AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt:153]: Can you remove the `toLong`? I tested locally and it semes to work without it

**Review by Arthur-Milchior — CHANGES_REQUESTED** [review]

**Comment by Arthur-Milchior** [comment]
Once this one is merged, finish #11388

**Comment by n1snt** [comment]
@Arthur-Milchior Just made all the changes you request please have a look at this & let me know if there are any changes to be made.

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt [hunk 2]
```diff
@@ -13,223 +13,255 @@
...
+        val manuallyBuriedCard = buryManually(sched, toAnswer.id)
+
+        // perform the search
+        val buriedCards = Finder(col).findCards(searchQuery, NoOrdering())
+
+        // assert
+        assertThat(
+            "A manually buried card should be returned",
+            buriedCards,
+            Matchers.hasItem(manuallyBuriedCard.id)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt:69]: import: `Matchers`

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt [hunk 3]
```diff
@@ -13,223 +13,255 @@
...
+import org.hamcrest.Matchers
+import org.hamcrest.core.Is
+import org.junit.Assert.*
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+import timber.log.Timber
+import java.util.*
+
+@RunWith(AndroidJUnit4::class)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt:44]: `@KotlinCleanup("is -> equalTo")`

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt [hunk 4]
```diff
@@ -13,223 +13,255 @@
...
+import android.util.Pair
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.ichi2.anki.CardBrowser
+import com.ichi2.anki.RobolectricTest
+import com.ichi2.anki.exception.ConfirmModSchemaException
+import com.ichi2.libanki.Consts.CARD_TYPE_REV
+import com.ichi2.libanki.Consts.QUEUE_TYPE_REV
+import com.ichi2.libanki.Consts.QUEUE_TYPE_SUSPENDED
+import com.ichi2.libanki.SortOrder.NoOrdering
+import com.ichi2.libanki.SortOrder.UseCollectionOrdering
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt:28]: Remove both these imports and qualify the callers with `SortOrder.`

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt [hunk 5]
```diff
@@ -238,236 +270,237 @@ public class FinderTest extends RobolectricTest {
...
-        assertEquals(4, col.findCards("tag:cat*").size());
-        assertEquals(2, col.findCards("tag:cat1").size());
-        assertEquals(2, col.findCards("tag:cat2").size());
-        assertEquals(1, col.findCards("tag:cat1::some").size());
-        assertEquals(2, col.findCards("tag:cat1::some*").size());
-        assertEquals(1, col.findCards("tag:cat1::something").size());
-        assertEquals(2, col.findCards("tag:cat2::some").size());
-        assertEquals(1, col.findCards("tag:cat2::some::").size());
+    fun test_findCardsHierarchyTag() {
+        val col = col
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt:383]: would be nice to fix the spacing of this method to match the Java

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt [hunk 6]
```diff
@@ -238,236 +270,237 @@ public class FinderTest extends RobolectricTest {
...
-        note.model().put("did", currentDid);
-        col.addNote(note);
-        long did = note.firstCard().getDid();
-        assertEquals(currentDid, did);
-        CardBrowser cb = super.startActivityNormallyOpenCollectionWithIntent(CardBrowser.class, new Intent());
-        cb.mDeckSpinnerSelection.updateDeckPosition(currentDid);
-        advanceRobolectricLooperWithSleep();
-        assertEquals(1L, cb.getCardCount());
+    fun test_deckNameContainingWildcardCanBeSearched() {
+        val `val` = "*Yr1::Med2::CAS4::F4: Renal::BRS (zanki)::HY"
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt:418]: `@KotlinCleanup("rename val")`

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt [hunk 7]
```diff
@@ -13,223 +13,257 @@
...
-        assertThat("sibling and manually buried should be the only cards returned", buriedCards, hasSize(2));
+    @Throws(
+        ConfirmModSchemaException::class
+    )
+    fun searchForBuriedReturnsManuallyAndSiblingBuried() {
+        val searchQuery = "is:buried"
+        val sched = upgradeToSchedV2() // needs to be first
+        enableBurySiblings()
+        super.addNoteUsingModelName("Basic (and reversed card)", "Front", "Back")
+        val toAnswer: Card? = sched.card
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt]: Don't refactor code in a migration commit, this should be as close to the Java as possible

**Review by david-allison — DISMISSED** [review]
Thanks!

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt (hunk 5 above)
Comment (n1snt) [reply AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt:384]: Done

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt (hunk 3 above)
Comment (n1snt) [reply AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt:44]: Done

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt (hunk 2 above)
Comment (n1snt) [reply AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt]: Done

File: AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt [hunk 8]
```diff
@@ -13,223 +13,255 @@
...
+        enableBurySiblings()
+        super.addNoteUsingModelName("Basic (and reversed card)", "Front", "Back")
+        val toAnswer: Card = sched.card!!
+
+        // act
+        val siblingBuried = burySiblings(sched, toAnswer)
+        val manuallyBuriedCard = buryManually(sched, toAnswer.id)
+
+        // perform the search
+        val buriedCards = Finder(col).findCards(searchQuery, NoOrdering())
```
[the remark this answers is not in the document]
Comment (n1snt) [reply AnkiDroid/src/test/java/com/ichi2/libanki/FinderTest.kt]: Done

**Review by Arthur-Milchior — APPROVED** [review]

**Pull request merged**